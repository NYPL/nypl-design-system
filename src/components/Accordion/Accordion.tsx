import * as React from "react";
import bem from "../../utils/bem";
import Icon from "../Icons/Icon";
import { IconNames } from "../Icons/IconTypes";

export interface AccordionProps {
  /** Inner label on the button that opens the accordion */
  accordionLabel?: React.ReactNode;
  /** BlockName for use with BEM. See how to work with blockNames and BEM here: http://getbem.com/introduction/ */
  blockName?: string;
  /** className you can add in addition to 'input' */
  className?: string;
  /** Whether the accordion is open by default */
  defaultOpen?: boolean;
  /** ID that other components can cross reference for accessibility purposes */
  id?: string;
  /** accordionLabel's input ID */
  inputId?: string;
  /** Modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
  modifiers?: string[];
}

/** Accordion component that shows content on toggle */
export default function Accordion(
  props: React.PropsWithChildren<AccordionProps>
) {
  const {
    modifiers = [],
    blockName,
    id,
    className,
    inputId,
    defaultOpen = false,
    accordionLabel,
    children,
  } = props;

  return (
    <div
      id={id}
      className={bem("accordion", modifiers, blockName, [className])}
    >
      <input
        id={`accordion-${inputId}`}
        className={bem("input", modifiers, "accordion")}
        type="checkbox"
        defaultChecked={defaultOpen}
      />
      <label
        htmlFor={`accordion-${inputId}`}
        className={bem("label", modifiers, "accordion")}
      >
        {accordionLabel}
        <Icon
          decorative={true}
          name={IconNames.minus}
          modifiers={["small", `${IconNames.minus}`]}
        />
        <Icon
          decorative={true}
          name={IconNames.plus}
          modifiers={["small", `${IconNames.plus}`]}
        />
      </label>

      <div className={bem("content", modifiers, "accordion")}>{children}</div>
    </div>
  );
}
