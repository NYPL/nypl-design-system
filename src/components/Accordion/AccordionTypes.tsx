export enum ColorVariantsAccordion {
  StatusNull = "statusNull",
  StatusDefault = "statusDefault",
  StatusWarning = "statusWarning",
  StatusError = "statusError",
}
