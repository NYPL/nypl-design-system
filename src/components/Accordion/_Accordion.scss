.accordion {
  border: 1px solid var(--ui-gray-medium);
  color: inherit;
  width: 100%;

  &--fixed-height {
    display: flex;
    flex-direction: column;
    max-height: inherit;
  }

  &__content {
    display: none;
    padding: var(--space-xxs) var(--space-xxxl) var(--space-xs) var(--space-s);
    transition: all 0.35s;
  }

  &__label {
    @include button--outline;

    align-items: center;
    border: none;
    cursor: pointer;
    font-weight: 500;
    justify-content: space-between;

    .icon--minus {
      display: none;
    }

    &:focus {
      position: relative;
    }

    &:hover {
      background-color: var(--ui-gray-light);
    }

    &--faq {
      font-size: var(--font-size-1);
      padding: var(--space-s);
    }
  }

  &__input {
    opacity: 0;
    position: absolute;
    z-index: -1;

    &:focus {
      + .accordion__label {
        @include focus-outline;
      }
    }

    &:checked {
      + .accordion__label .icon--minus {
        display: inline;
      }

      + .accordion__label .icon--plus {
        display: none;
      }

      ~ .accordion__content {
        display: inline-block;

        &--fixed-height {
          // Adds margin so that content always appears outside the focus ring
          margin-top: var(--space-xxs);
          overflow-y: auto;
        }
      }
    }
  }

  // Eliminates the double border on accordion sets in ul-lis
  li:not(:first-of-type) & {
    margin-top: -1px;
  }
}
