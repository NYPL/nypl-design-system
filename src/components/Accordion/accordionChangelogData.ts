/** This data is used to populate the ComponentChangelogTable component.
 *
 * date: string (when adding new entry during development, set value as "Prerelease")
 * version: string (when adding new entry during development, set value as "Prerelease")
 * type: "Bug Fix" | "New Feature" | "Update";
 * affects: array["Accessibility" | "Documentation" | "Functionality" | "Styles"];
 * notes: array (will render as a bulleted list, add one array element for each list element)
 */
import { ChangelogData } from "../../utils/ComponentChangelogTable";

export const changelogData: ChangelogData[] = [
  {
    date: "Prerelease",
    version: "Prerelease",
    type: "Update",
    affects: ["Accessibility", "Functionality"],
    notes: [
      "Adds logic to close accordion when accordion button is focused and 'esc' key is pressed",
      "Adds logic to close accordion when element within panel is focused and 'esc' key is pressed",
    ],
  },
  {
    date: "2024-03-14",
    version: "3.0.0",
    type: "Update",
    affects: ["Accessibility", "Functionality"],
    notes: [
      "Chakra 2.8 update.",
      "Added the ariaLabel and buttonInteractionRef prop.",
    ],
  },
];
