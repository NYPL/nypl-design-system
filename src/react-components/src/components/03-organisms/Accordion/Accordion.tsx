import * as React from "react";
import Button, { ButtonOptions } from "../../01-atoms/Button/Button";
import bem from "../../../utils/bem";
import { ButtonTypes, ButtonIconPositions } from "../../01-atoms/Button/ButtonTypes";

export interface AccordionProps {
  id: string;

  /** Optional modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
  modifiers?: string[];
  /** Optional blockName for use with BEM. See how to work with blockNames and BEM here: http://getbem.com/introduction/ */
  blockName?: string;

  /** The button that controls the acccordion toggle */
  buttonOptions: ButtonOptions;
}

/** Accordion component that shows content on toggle */

export default class Accordion extends React.Component<AccordionProps, { isOpen: boolean }> {
  constructor(props: AccordionProps) {
    super(props);
    this.state = {
      isOpen: false
    };
    this.toggleContentShow = this.toggleContentShow.bind(this);
  }

  toggleContentShow() {
    this.setState(prevState => ({ isOpen: !prevState.isOpen }));
  }

  render() {
    const { buttonOptions, modifiers = [], blockName = "" } = this.props;
    buttonOptions.type = "button";
    buttonOptions.blockName = "accordion";
    buttonOptions.modifiers = ["large"];
    buttonOptions.iconPosition = ButtonIconPositions.JustifyRight;
    buttonOptions.iconName = this.state.isOpen ? "minus" : "plus";
    buttonOptions.buttonType = ButtonTypes.Outline;

    return (
      <div className={bem("accordion", modifiers, blockName)}>
        <Button callback={this.toggleContentShow} {...buttonOptions}>{buttonOptions.content}</Button>
        {this.state.isOpen && <div className="accordion-content">
          {this.props.children}
        </div>}
      </div >);
  }
}

