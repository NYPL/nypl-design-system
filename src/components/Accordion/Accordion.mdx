import { ArgTypes, Canvas, Description, Meta, Source } from "@storybook/blocks";
import ComponentChangelogTable from "../../utils/ComponentChangelogTable";
import { changelogData } from "./accordionChangelogData";

import * as AccordionStories from "./Accordion.stories";
import Link from "../Link/Link";

<Meta of={AccordionStories} />

# Accordion

| Component Version | DS Version   |
| ----------------- | ------------ |
| Added             | `0.1.0`      |
| Latest            | `Prerelease` |

## Table of Contents

- {<Link href="#overview" target="_self">Overview</Link>}
- {<Link href="#component-props" target="_self">Component Props</Link>}
- {<Link href="#accessibility" target="_self">Accessibility</Link>}
- {<Link href="#faq-example" target="_self">FAQ Example</Link>}
- {<Link href="#example-with-panelmaxheight-prop" target="_self">Example with panelMaxHeight Prop</Link>}
- {<Link href="#changelog" target="_self">Changelog</Link>}

## Overview

<Description of={AccordionStories} />

The `Accordion` component displays a list of high-level options that can
expand and collapse to reveal associated sections of content. This component
pushes existing content on the page down. Each accordion item is self contained,
but when the data for multiple accordions is passed in through the `accordionData`
prop, the `Accordion` components are grouped together and rendered under a single
element. When grouped like this, each `Accordion` component still opens and
closes independent from the others in the group.

The only way to render an `Accordion` component is to pass in an array of objects
with `label`, `panel`, and optional `accordionType` properties for each accordion item.
Note that you can pass in a string or DOM elements into the `panel` property in each
object. This approach is needed because, internally, we deal with the logic to render
the necessary icon, Chakra components, and styles.

The `accordionType` prop can be used to change the accordion button's background color.
It takes a value of `"default"` for `"ui.white"`, `"warning"` for `"ui.status.primary"`,
and `"error"` for `"ui.status.secondary"`. The optional `panelMaxHeight` prop determines
the maximum height of the `Accordion`'s panel when expanded. If the content within the
panel exceeds the `panelMaxHeight`, a scrollbar will appear.

<Source
  code={`
const accordionData = [
  {
    accordionType: "default",
    label: "Tom Nook",
    panel: (
      <Card
        imageProps={{
          alt: "Alt text",
          aspectRatio: "twoByOne",
          src: "https://play.nintendo.com/images/AC_Tom_FRYtwIN.17345b1513ac044897cfc243542899dce541e8dc.9afde10b.png",
        }}
        isCentered
        layout="row"
      >
        <CardHeading level="h4" id="heading1">
          Tom Nook
        </CardHeading>
        <CardContent>
          Tom Nook, <b>known in Japan as Tanukichi</b>, is a fictional
          character in the Animal Crossing series who operates the
          village store.
        </CardContent>
      </Card>
    ),
  },
];

...

<Accordion accordionData={accordionData} />
`} language="tsx" />

## Component Props

<Canvas of={AccordionStories.WithControls} />

<ArgTypes of={AccordionStories.WithControls} />

## Accessibility

- Follows [U.S. Web Design System (USWDS)](https://designsystem.digital.gov/components/accordion/)
  guidelines:
  - Use the `button` HTML element for screenreaders and keyboards.
  - Use the `aria-expanded` attribute to indicate whether the `Accordion` is
    expanded (`true`) or collapsed (`false`).
  - Use the `aria-controls` attribute to associate the control with the panel.
- Deviates from USWDS in that the user can open multiple panels. Opening one,
  does not collapse already expanded panel.
- The open and close icons are decorative (`aria-hidden` is `true`).
- Visible focus goes around full button and full button is clickable.

Resources:

- [W3C WAI-Aria Practices - Accordion](https://www.w3.org/TR/wai-aria-practices-1.1/#accordion)
- [USWDS Accordion](https://designsystem.digital.gov/components/accordion/)
- [Chakra Accordion](https://chakra-ui.com/docs/components/disclosure/accordion)

## FAQ Example

Building out FAQ-like accordions happens automatically when there are more than
one object in the array passed into the `accordionData` prop.

<Canvas of={AccordionStories.FaqContent} />

<Source
  code={`
export const faqContent: AccordionDataProps[] = [
  {
    accordionType: "default",
    label: "Tom Nook",
    panel: (
      <Card
        imageProps={{
          alt: "Alt text",
          aspectRatio: "twoByOne",
          src: "https://play.nintendo.com/images/AC_Tom_FRYtwIN.17345b1513ac044897cfc243542899dce541e8dc.9afde10b.png",
        }}
        isCentered
        layout="row"
      >
        <CardHeading level="h4" id="heading1-tom">
          Tom Nook
        </CardHeading>
        <CardContent>
          Tom Nook, <b>known in Japan as Tanukichi</b>, is a fictional character
          in the Animal Crossing series who operates the village store.
        </CardContent>
      </Card>
    ),
  },
  {
    accordionType: "warning",
    label: "Isabelle",
    panel: (
      <Card
        imageProps={{
          alt: "Alt text",
          aspectRatio: "twoByOne",
          src: "https://play.nintendo.com/images/AC_Isabelle_7XU6aGu.17345b1513ac044897cfc243542899dce541e8dc.9afde10b.png",
        }}
        isCentered
        layout="row"
      >
        <CardHeading level="h4" id="heading1-isabelle">
          Isabelle
        </CardHeading>
        <CardContent>
          Isabelle, known as Shizue in Japan, is a fictional character from the
          Animal Crossing series of video games. She is a kindly Shih Tzu that
          debuted in the 2012 release Animal Crossing: New Leaf, where she
          serves as the secretary to the player character.
        </CardContent>
      </Card>
    ),
  },
  {
    accordionType: "error",
    label: "K.K. Slider",
    panel: (
      <Card
        imageProps={{
          alt: "Alt text",
          aspectRatio: "twoByOne",
          src: "https://play.nintendo.com/images/AC_KK_jh4yj5t.17345b1513ac044897cfc243542899dce541e8dc.9afde10b.png",
        }}
        isCentered
        layout="row"
      >
        <CardHeading level="h4" id="heading1-kkslider">
          K.K. Slider
        </CardHeading>
        <CardContent>
          <p>
            Totakeke, more commonly known as <b>K.K. Slider or K.K.</b>, is a
            fictional character within the Animal Crossing franchise. One of the
            franchise's most popular characters, he debuted in the title Animal
            Crossing, and has appeared in every installment since.
          </p>
        </CardContent>
      </Card>
    ),
  },
];

...

<Accordion accordionData={faqContent} />
`} language="tsx" />

## Examples with panelMaxHeight Prop

Example with image and text

<Canvas of={AccordionStories.PanelMaxHeightExample} />

Example with multiple paragraphs

<Canvas of={AccordionStories.PanelMaxHeightExample2} />

## Always Render the Accordion Content

By default, the content in an `Accordion` is only rendered when it is opened. When
it is closed, the content gets removed from the DOM. This is an intentional
optimization. However, there will be occasions when you want to always render the
content inside an `Accordion` whether it's opened or closed. Use the
`isAlwaysRendered` prop to always render the `Accordion`'s content.

In the following examples, the first `Accordion` has the default behavior where
the content is removed when it is closed. The second `Accordion` always renders
its content. This is important in this scenario because we want to keep the state
of the content when it is closed. Click on subjects inside both `Accordion`s and
toggle them closed and opened to see the difference.

<Canvas of={AccordionStories.IsAlwaysRendered} />

## Changelog

<ComponentChangelogTable changelogData={changelogData} />
