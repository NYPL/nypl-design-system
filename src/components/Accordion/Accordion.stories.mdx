import {
  Meta,
  Story,
  ArgsTable,
  Preview,
  Description,
} from "@storybook/addon-docs/blocks";
import Accordion from "./Accordion";
import { withDesign } from "storybook-addon-designs";
import * as stories from "./Accordion.stories.tsx";
import { list as ListStory } from "../List/List.stories.tsx";
import { Source } from "@storybook/addon-docs/blocks";
import dedent from "ts-dedent";
import ReactDOMServer from "react-dom/server";

<Meta
  title="Accordion"
  component={Accordion}
  decorators={[withDesign]}
  parameters={{
    design: {
      type: "figma",
      url:
        "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Master?node-id=10734%3A2520",
    },
  }}
  argTypes={{
    children: { table: { disable: true } },
  }}
/>

# Accordion

<Description of={Accordion} />

The Accordion component pushes existing content on the page down. The
implementation is built using basic input/label DOM elements and CSS.
Therefore, make sure to pass the `inputId` prop to associate the label with
the input element. If multiple Accordion components are used on a page, make
sure that each `inputId` is unique.

<Preview withToolbar>
  <Story
    name="Basic"
    args={{
      accordionLabel: "Click to expand",
      children: <ListStory {...ListStory.args} />,
      inputId: "accordionBtn",
    }}
  >
    {args => <Accordion {...args} />}
  </Story>
</Preview>

<ArgsTable story="Basic" />

## FAQ example

The Accordion component has an optional `modifiers` prop that has pre-styled
designs. For example, when building out FAQ-like pages, pass
`modifiers=["faq"]` into the Accordion component.

<Preview withToolbar>
  <Story story={stories.AccordionAsFAQSet} />
</Preview>

## Scrollable example

Use modifiers=["fixed-height"] when you'd like the accordion height to match that of its
parent container. In this example, we wrap the accordion in a container whose height is
set to 300px. If the contents within the accordion are taller than the height of the parent
container, a scrollbar will appear.

As in the other examples, any page content underneath the accordion will continue to be pushed down.

<Preview withToolbar>
  <Story story={stories.AccordionScroll} />
</Preview>

<Source code={ReactDOMServer.renderToString(stories.AccordionScroll)} />
