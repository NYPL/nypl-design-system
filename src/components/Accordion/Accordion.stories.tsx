import * as React from "react";
import range from "lodash/range";

import Accordion from "./Accordion";
import List from "../List/List";
import { ListTypes } from "../List/ListTypes";
import Link from "../Link/Link";
import { list as ListStory } from "../List/List.stories";

// Set up the reusable template to create a list of Accordion components.
const AccordionListTemplate = ({ count, children, ...args }) => (
  <List type={ListTypes.Unordered} modifiers={["no-list-styling"]}>
    {range(count).map(i => (
      <li key={i}>
        <Accordion {...args} inputId={`${args.inputId}-${i}`}>
          {children}
        </Accordion>
      </li>
    ))}
  </List>
);

const faqContent = (
  <>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
      tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit
      amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
      labore et dolore magna aliqua. Lorem ipsum dolor sit amet, consectetur
      adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
      magna aliqua.
    </p>

    <p>
      <Link href="#">Learn More</Link>
    </p>
  </>
);

// This is one specific story where it will render a list of FAQs.
export const AccordionAsFAQSet = AccordionListTemplate.bind({});
AccordionAsFAQSet.args = {
  accordionLabel: "FAQ Question Lorem Ipsum",
  modifiers: ["faq"],
  inputId: "accordionBtn",
  children: faqContent,
  count: 3,
};

// One accordion with scrollable list
export const AccordionScroll = _props => (
  <>
    <div style={{ maxHeight: "300px", width: "500px" }}>
      <Accordion
        accordionLabel={"Content"}
        modifiers={["fixed-height"]}
        defaultOpen={true}
      >
        <ListStory {...ListStory.args}></ListStory>
      </Accordion>
    </div>
    <div>Example content underneath</div>
  </>
);

AccordionScroll.storyName = "Accordion with Scrollable Content";
