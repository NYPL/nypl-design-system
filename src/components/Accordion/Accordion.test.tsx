import { expect } from "chai";
import * as Enzyme from "enzyme";
import * as React from "react";

import Accordion from "./Accordion";

describe("Accordion", () => {
  let container;
  before(() => {
    container = Enzyme.mount(
      <Accordion
        id="accordion"
        inputId="accordionBtn"
        accordionLabel="Click to expand"
      >
        <p className="accordion-content">this is the accordion content</p>
      </Accordion>
    );
  });

  it("Renders an input checkbox and label", () => {
    expect(container.find("input").exists()).to.equal(true);
    expect(container.find(".accordion__label").text()).to.contain(
      "Click to expand"
    );
  });
});
