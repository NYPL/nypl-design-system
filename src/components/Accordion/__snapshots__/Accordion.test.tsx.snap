// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Renders the UI snapshot correctly 1`] = `
<div
  className="chakra-accordion css-1xdhyk6"
  id="accordian"
>
  <div
    className="chakra-accordion__item css-1fsnuue"
  >
    <button
      aria-controls="accordion-panel-accordian-item-0"
      aria-expanded={false}
      className="chakra-accordion__button css-1c6323r"
      disabled={false}
      id="accordion-button-accordian-item-0"
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <span
        className="css-ox82o1"
      >
        Gerry Kellman
      </span>
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="accordion-accordian-icon-0"
        role="img"
        title="plus icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </button>
    <div
      className="chakra-collapse"
      style={
        Object {
          "display": "none",
          "height": "0px",
          "opacity": 0,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-labelledby="accordion-button-accordian-item-0"
        className="chakra-accordion__panel css-0"
        id="accordion-panel-accordian-item-0"
        role="region"
      >
        <div
          className="css-1u8qly9"
          id="card"
        >
          <div
            className="css-0"
          >
            <div
              className="css-0"
            >
              <div
                className="the-wrap css-1u8qly9"
              >
                <div
                  className="the-crop css-0"
                />
              </div>
            </div>
          </div>
          <div
            className="card-body css-0"
          >
            <h4
              className="chakra-heading css-1xdhyk6"
              id="heading1"
              layout="row"
            >
              Gerry Kellman
            </h4>
            <div
              className="css-1xdhyk6"
            >
              Gerri is 
              <b>
                one of Logan's most trusted confidantes
              </b>
              , one who serves many roles within the company. She's one of the most powerful people at Waystar Royco outside of the family itself.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Accordion Renders the UI snapshot correctly 2`] = `
<div
  className="chakra-accordion css-1xdhyk6"
  id="accordian"
>
  <div
    className="chakra-accordion__item css-1fsnuue"
  >
    <button
      aria-controls="accordion-panel-accordian-item-0"
      aria-expanded={false}
      className="chakra-accordion__button css-1c6323r"
      disabled={false}
      id="accordion-button-accordian-item-0"
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <span
        className="css-ox82o1"
      >
        Gerry Kellman
      </span>
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="accordion-accordian-icon-0"
        role="img"
        title="plus icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </button>
    <div
      className="chakra-collapse"
      style={
        Object {
          "display": "none",
          "height": "0px",
          "opacity": 0,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-labelledby="accordion-button-accordian-item-0"
        className="chakra-accordion__panel css-0"
        id="accordion-panel-accordian-item-0"
        role="region"
      >
        <div
          className="css-1u8qly9"
          id="card"
        >
          <div
            className="css-0"
          >
            <div
              className="css-0"
            >
              <div
                className="the-wrap css-1u8qly9"
              >
                <div
                  className="the-crop css-0"
                />
              </div>
            </div>
          </div>
          <div
            className="card-body css-0"
          >
            <h4
              className="chakra-heading css-1xdhyk6"
              id="heading1"
              layout="row"
            >
              Gerry Kellman
            </h4>
            <div
              className="css-1xdhyk6"
            >
              Gerri is 
              <b>
                one of Logan's most trusted confidantes
              </b>
              , one who serves many roles within the company. She's one of the most powerful people at Waystar Royco outside of the family itself.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Accordion Renders the UI snapshot correctly 3`] = `
<div
  className="chakra-accordion css-1t0bvx9"
  id="accordian"
>
  <div
    className="chakra-accordion__item css-1fsnuue"
  >
    <button
      aria-controls="accordion-panel-accordian-item-0"
      aria-expanded={false}
      className="chakra-accordion__button css-1c6323r"
      disabled={false}
      id="accordion-button-accordian-item-0"
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <span
        className="css-ox82o1"
      >
        Gerry Kellman
      </span>
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="accordion-accordian-icon-0"
        role="img"
        title="plus icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </button>
    <div
      className="chakra-collapse"
      style={
        Object {
          "display": "none",
          "height": "0px",
          "opacity": 0,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-labelledby="accordion-button-accordian-item-0"
        className="chakra-accordion__panel css-0"
        id="accordion-panel-accordian-item-0"
        role="region"
      >
        <div
          className="css-1u8qly9"
          id="card"
        >
          <div
            className="css-0"
          >
            <div
              className="css-0"
            >
              <div
                className="the-wrap css-1u8qly9"
              >
                <div
                  className="the-crop css-0"
                />
              </div>
            </div>
          </div>
          <div
            className="card-body css-0"
          >
            <h4
              className="chakra-heading css-1xdhyk6"
              id="heading1"
              layout="row"
            >
              Gerry Kellman
            </h4>
            <div
              className="css-1xdhyk6"
            >
              Gerri is 
              <b>
                one of Logan's most trusted confidantes
              </b>
              , one who serves many roles within the company. She's one of the most powerful people at Waystar Royco outside of the family itself.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Accordion Renders the UI snapshot correctly 4`] = `
<div
  className="chakra-accordion css-1xdhyk6"
  data-testid="testid"
  id="accordian"
>
  <div
    className="chakra-accordion__item css-1fsnuue"
  >
    <button
      aria-controls="accordion-panel-accordian-item-0"
      aria-expanded={false}
      className="chakra-accordion__button css-1c6323r"
      disabled={false}
      id="accordion-button-accordian-item-0"
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <span
        className="css-ox82o1"
      >
        Gerry Kellman
      </span>
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="accordion-accordian-icon-0"
        role="img"
        title="plus icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </button>
    <div
      className="chakra-collapse"
      style={
        Object {
          "display": "none",
          "height": "0px",
          "opacity": 0,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-labelledby="accordion-button-accordian-item-0"
        className="chakra-accordion__panel css-0"
        id="accordion-panel-accordian-item-0"
        role="region"
      >
        <div
          className="css-1u8qly9"
          id="card"
        >
          <div
            className="css-0"
          >
            <div
              className="css-0"
            >
              <div
                className="the-wrap css-1u8qly9"
              >
                <div
                  className="the-crop css-0"
                />
              </div>
            </div>
          </div>
          <div
            className="card-body css-0"
          >
            <h4
              className="chakra-heading css-1xdhyk6"
              id="heading1"
              layout="row"
            >
              Gerry Kellman
            </h4>
            <div
              className="css-1xdhyk6"
            >
              Gerri is 
              <b>
                one of Logan's most trusted confidantes
              </b>
              , one who serves many roles within the company. She's one of the most powerful people at Waystar Royco outside of the family itself.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Accordion Renders the UI snapshot correctly 5`] = `
<div
  className="chakra-accordion css-1xdhyk6"
  id="accordian"
>
  <div
    className="chakra-accordion__item css-1fsnuue"
  >
    <button
      aria-controls="accordion-panel-accordian-item-0"
      aria-expanded={false}
      className="chakra-accordion__button css-1c8csqn"
      disabled={false}
      id="accordion-button-accordian-item-0"
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <span
        className="css-ox82o1"
      >
        Gerry Kellman
      </span>
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="accordion-accordian-icon-0"
        role="img"
        title="plus icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </button>
    <div
      className="chakra-collapse"
      style={
        Object {
          "display": "none",
          "height": "0px",
          "opacity": 0,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-labelledby="accordion-button-accordian-item-0"
        className="chakra-accordion__panel css-0"
        id="accordion-panel-accordian-item-0"
        role="region"
      >
        <div
          className="css-1u8qly9"
          id="card"
        >
          <div
            className="css-0"
          >
            <div
              className="css-0"
            >
              <div
                className="the-wrap css-1u8qly9"
              >
                <div
                  className="the-crop css-0"
                />
              </div>
            </div>
          </div>
          <div
            className="card-body css-0"
          >
            <h4
              className="chakra-heading css-1xdhyk6"
              id="heading1"
              layout="row"
            >
              Gerry Kellman
            </h4>
            <div
              className="css-1xdhyk6"
            >
              Gerri is 
              <b>
                one of Logan's most trusted confidantes
              </b>
              , one who serves many roles within the company. She's one of the most powerful people at Waystar Royco outside of the family itself.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Accordion Renders the UI snapshot correctly 6`] = `
<div
  className="chakra-accordion css-1xdhyk6"
  id="accordian"
>
  <div
    className="chakra-accordion__item css-1fsnuue"
  >
    <button
      aria-controls="accordion-panel-accordian-item-0"
      aria-expanded={false}
      className="chakra-accordion__button css-1lty5mh"
      disabled={false}
      id="accordion-button-accordian-item-0"
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <span
        className="css-ox82o1"
      >
        Gerry Kellman
      </span>
      <svg
        aria-hidden={true}
        className="chakra-icon css-1grhd2q"
        focusable={false}
        id="accordion-accordian-icon-0"
        role="img"
        title="plus icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </button>
    <div
      className="chakra-collapse"
      style={
        Object {
          "display": "none",
          "height": "0px",
          "opacity": 0,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-labelledby="accordion-button-accordian-item-0"
        className="chakra-accordion__panel css-0"
        id="accordion-panel-accordian-item-0"
        role="region"
      >
        <div
          className="css-1u8qly9"
          id="card"
        >
          <div
            className="css-0"
          >
            <div
              className="css-0"
            >
              <div
                className="the-wrap css-1u8qly9"
              >
                <div
                  className="the-crop css-0"
                />
              </div>
            </div>
          </div>
          <div
            className="card-body css-0"
          >
            <h4
              className="chakra-heading css-1xdhyk6"
              id="heading1"
              layout="row"
            >
              Gerry Kellman
            </h4>
            <div
              className="css-1xdhyk6"
            >
              Gerri is 
              <b>
                one of Logan's most trusted confidantes
              </b>
              , one who serves many roles within the company. She's one of the most powerful people at Waystar Royco outside of the family itself.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
